### PR TITLE
fix(cluster_aws.py): Add logging for creating an on demand instance

### PR DESCRIPTION
**Pull request by yarongilor**
In order to verify correctness of the instance-create-request.
	
Refs:  [#5645](https://github.com/scylladb/scylla-cluster-tests/issues/5645)
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
Tested in: https://jenkins.scylladb.com/job/scylla-staging/job/yarongilor/job/artifacts-ami-test-add-logs/

Example output:
```
11:18:20  < t:2023-01-08 09:18:20,008 f:cluster_aws.py  l:149  c:sdcm.cluster_aws     p:INFO  > Using EC2 service with DC-index: 0, (associated with region: us-east-1)
11:18:20  < t:2023-01-08 09:18:20,008 f:cluster_aws.py  l:151  c:sdcm.cluster_aws     p:INFO  > Using Availability Zone of: us-east-1a
11:18:20  < t:2023-01-08 09:18:20,008 f:cluster_aws.py  l:152  c:sdcm.cluster_aws     p:INFO  > Sending an On-Demand request with params: {'ImageId': 'ami-02643ff1876ab4973', 'UserData': 'Content-Type: multipart/mixed; boundary="===============8547825398890058103=="\nMIME-Version: 1.0\n\n--===============8547825398890058103==\nContent-Type: x-scylla/json\nMIME-Version: 1.0\nContent-Disposition: attachment; filename="scylla_machine_image.json"\n\n{\n    "cluster_name": "artifacts-ami-jenkins-db-cluster-53b44bd3",\n    "data_device": "instance_store",\n    "raid_level": 0,\n    "scylla_yaml": {\n        "cluster_name": "artifacts-ami-jenkins-db-cluster-53b44bd3"\n    },\n    "start_scylla_on_first_boot": false\n}\n--===============8547825398890058103==\nContent-Type: text/cloud-config\nMIME-Version: 1.0\nContent-Disposition: attachment; filename="cloud-config.txt"\n\n\n        #cloud-config\n        cloud_final_modules:\n        - [scripts-user, always]\n        \n--===============8547825398890058103==\nContent-Type: text/x-shellscript\nMIME-Version: 1.0\nContent-Disposition: attachment; filename="user-script.txt"\n\n#!/bin/bash\nset -x\nwhile ! systemctl status cloud-init.service | grep "active (exited)"; do sleep 1; done\nsystemctl stop sshd || true\n\nif [ -f "/etc/security/limits.d/20-nproc.conf" ]; then\n    sed -i -e "s/^\\*[[:blank:]]*soft[[:blank:]]*nproc[[:blank:]]*.*/*\t\tsoft\tnproc\t\tunlimited/"     /etc/security/limits.d/20-nproc.conf || true\nelse\n    echo "*    hard    nproc    unlimited" > /etc/security/limits.d/20-nproc.conf || true\nfi\n\nsed -i "s/#MaxSessions \\(.*\\)$/MaxSessions 1000/" /etc/ssh/sshd_config || true\nsed -i "s/#MaxStartups \\(.*\\)$/MaxStartups 60/" /etc/ssh/sshd_config || true\nsed -i "s/#LoginGraceTime \\(.*\\)$/LoginGraceTime 15s/" /etc/ssh/sshd_config || true\n\nif (( $(ssh -V 2>&1 | tr -d "[:alpha:][:blank:][:punct:]" | cut -c-2) >= 88 )); then\n    systemctl stop sshd || true\n    sed -i "s/#PubkeyAuthentication \\(.*\\)$/PubkeyAuthentication yes/" /etc/ssh/sshd_config || true\n    sed -i -e "\\$aPubkeyAcceptedAlgorithms +ssh-rsa" /etc/ssh/sshd_config || true\n    sed -i -e "\\$aHostKeyAlgorithms +ssh-rsa" /etc/ssh/sshd_config || true\n    systemctl restart sshd || true\nfi\nsystemctl restart sshd || true\nsystemctl start sshd || true\n\n--===============8547825398890058103==--\n', 'MinCount': 1, 'MaxCount': 1, 'KeyName': 'scylla-qa-ec2', 'BlockDeviceMappings': [{'DeviceName': '/dev/sda1', 'Ebs': {'VolumeSize': 50, 'VolumeType': 'gp2'}}], 'NetworkInterfaces': [{'DeviceIndex': 0, 'SubnetId': 'subnet-0a09ba4421ec6aaa8', 'AssociatePublicIpAddress': True, 'Groups': ['sg-0feef3370ee8305ac']}], 'InstanceType': 'i3.2xlarge', 'IamInstanceProfile': {'Name': 'qa-scylla-manager-backup-instance-profile'}}
```

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -143,9 +143,16 @@ def _create_on_demand_instances(self, count, interfaces, ec2_user_data, dc_idx=0
         instance_profile = self.instance_profile_name
         if instance_profile:
             params['IamInstanceProfile'] = {'Name': instance_profile}
+        ec2 = ec2_client.EC2ClientWrapper(region_name=self.region_names[dc_idx])
+        subnet_info = ec2.get_subnet_info(self._ec2_subnet_id[dc_idx])
+        region_name_with_az = subnet_info['AvailabilityZone']
+        LOGGER.info('Sending an On-Demand request with params: %s', params)
```
Comment (fruch) [inline sdcm/cluster_aws.py:149]: the params has too much information, please put his print in debug level as well

Comment (yarongilor) [reply sdcm/cluster_aws.py]: fixed

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**